src/windows/main_window.py

```diff
@@ -2782,6 +2783,9 @@ def __init__(self, mode=None):
         # Save settings
         s.save()
 
+        # Last window state to restore from full screen
+        self.last_window_state = self.windowState()
+
```

Comment:
The problem with storing this in the application is, it can get out of sync with reality pretty easily. Unless you record *every* state change that can occur in the application both while it's fullscreened and while it's not.
```suggestion
```

Reply:
It intended to store only window state before going into fullscreen (to return normally). No need to store other states, because they all saved in settings already.